use of undeclared identifier 'asm'  (c11)

I'm working on support for alternate C compilers (zig, llvm) and cross-compilation using Bazel. I came across this minor problem when compiling with `--std=c11`:

```
runtime/caml/platform.h:40:3: error: use of undeclared identifier 'asm'
  asm volatile("pause" ::: "memory");
  ^
```

The fix is easy, just use `__asm__` instead of `asm`.  See [Alternate Keywords](https://gcc.gnu.org/onlinedocs/gcc-7.2.0/gcc/Alternate-Keywords.html#Alternate-Keywords).

--- a/runtime/caml/platform.h
+++ b/runtime/caml/platform.h
@@ -36,12 +36,12 @@
 Caml_inline void cpu_relax() {
 #ifdef __GNUC__
 #if defined(__x86_64__) || defined(__i386__)
-  asm volatile("pause" ::: "memory");
+  __asm__ volatile("pause" ::: "memory");
 #elif defined(__aarch64__)
-  asm volatile ("yield" ::: "memory");
+  __asm__ volatile ("yield" ::: "memory");
 #else
   /* Just a compiler barrier */
-  asm volatile ("" ::: "memory");
+  __asm__ volatile ("" ::: "memory");
 #endif
 #endif
 }

